sup3r/qa/feature_plots.py

```diff
@@ -0,0 +1,211 @@
+                out_file = self.out_pattern.format(feature=feature,
+                                                   index=i)
+                self.plot_figure(res, time_index, feature, t_slice,
+                                 out_file)
+
+    def _parallel_figure_plots(self, res, time_index, time_slices):
+        """Plot figures in parallel with max_workers=self.workers
+
+        Parameters
+        ----------
+        res : MultiFileResourceX
+            Resource handler for the provided h5 files
+        time_index : pd.DateTimeIndex
+            The time index for the provided h5 files
+        time_slices : list
+             List of slices specifying all the time ranges to average and plot
+        """
+        futures = {}
+        now = dt.now()
+        n_files = len(time_slices) * len(self.features)
+        with ThreadPoolExecutor(max_workers=self.workers) as exe:
+            for feature in self.features:
+                for i, t_slice in enumerate(time_slices):
+                    out_file = self.out_pattern.format(feature=feature,
+                                                       index=i)
+                    future = exe.submit(self.plot_figure, res, time_index,
+                                        feature, t_slice, out_file)
+                    futures[future] = out_file
+
+            logger.info(f'Started plotting {n_files} files '
+                        f'in {dt.now() - now}.')
+
+            for i, future in enumerate(as_completed(futures)):
+                try:
+                    future.result()
+                except Exception as e:
+                    msg = (f'Error making plot {futures[future]}.')
+                    logger.exception(msg)
+                    raise RuntimeError(msg) from e
+                logger.debug(f'{i+1} out of {n_files} plots created.')
+
+    def plot_figure(self, res, time_index, feature, t_slice, out_file):
+        """Plot temporal average for the given feature and with the time range
+        specified by t_slice
+
+        Parameters
+        ----------
+        res : MultiFileResourceX
+            Resource handler for the provided h5 files
+        time_index : pd.DateTimeIndex
+            The time index for the provided h5 files
+        feature : str
+            The feature to plot
+        t_slice : slice
+            The slice specifying the time range to average and plot
+        out_file : str
+            Name of the output plot file
+        """
+        start_time = time_index[t_slice.start]
+        stop_time = time_index[t_slice.stop - 1]
+        logger.info(f'Plotting time average for {feature} from '
+                    f'{start_time} to {stop_time}.')
+        fig = plt.figure()
+        title = f'{feature}: {start_time} - {stop_time}'
+        plt.suptitle(title)
+        plt.scatter(res.meta.longitude, res.meta.latitude,
+                    c=np.mean(res[feature][t_slice], axis=0), **self.kwargs)
```

Comment:
Recommendations:

1. Add option for a spatial slice that would plot at an interval. For 2km over CONUS you can often get away with plotting every 5 or 10 pixels which speeds things up significantly. 
2. The extraction from the rex resource handlers should look like `res[feature, t_slice, s_slice]` where what you have `res[feature][t_slice]` will actually retrieve the FULL feature array and then subsequently slice it in memory. This has been a big debate because xarray and h5py use the syntax you're using here but it's too big of a refactor right now given how widely rex is being used across the lab.

Reply:
Ah yeah, keep using the xarray syntax by default. Good catch.